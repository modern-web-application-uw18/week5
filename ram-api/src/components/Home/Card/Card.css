.Card {
    height: 100px;
    width: 100px;
    display: grid;
    justify-content: center;
    margin: 5px;
    grid-template-columns: 49% 50%;
    grid-auto-rows: 49%;
    grid-gap: 5px;
    background-position: 50% 50%;
    background-size: cover;
    text-align: left;
    padding: 5px;
    border: solid 1px #cecece;
    background-color: beige;
    transition: background-color  3s;
    transition-timing-function: ease;
}

.Card:hover,
.Card:active,
.Card.active {
    background-color: #d6d6d6;
    transition: background-color  0.4s;
    cursor: pointer;
    transition-timing-function: ease;
}