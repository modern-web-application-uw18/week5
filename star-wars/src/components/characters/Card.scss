.character-list {
  text-align: center;
  display: block;
  margin-bottom: 1em;
}
