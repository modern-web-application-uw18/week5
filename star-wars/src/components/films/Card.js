import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class Card extends Component {

  render() {
    const url = `films/${this.props.data.url.split('/')[5]}`;

    return (
      <div>
        <Link className='character-list' to={url}>{this.props.data.title}</Link>
      </div>
    );
  }
}

Card.propTypes = {
  data: PropTypes.shape({
      episode_id: PropTypes.number,
      release_date: PropTypes.string,
      title: PropTypes.string
  })
};

export default Card;
