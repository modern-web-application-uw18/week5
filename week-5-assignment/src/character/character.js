import React, {Component} from 'react';
import '../App.css'
import 'bootstrap/dist/css/bootstrap.css';
import PropTypes from 'prop-types';

class Character extends Component{
    constructor(props){
        super(props);
        this.state = {
            character: this.character
        };
    }

    character = this.props.match.params.number;

    componentDidMount(){
        fetch(`https://swapi.co/api/people/?search=${this.character}`)
        .then(response => response.json())
        .then(json => this.setState((prevState) => {
            return {character: json.results};
    }));
    }

    

    render(){
    return (
      <div className="border border-success bg-light">
        <p>Name: {this.state.character[0].name}</p>
        <p>Height: {this.state.character[0].height}</p>
        <p>Mass: {this.state.character[0].mass}</p>
        <p>Birth Year: {this.state.character[0].birth_year}</p>
        <p>Gender: {this.state.character[0].gender}</p>
      </div>
    )
  }
}

Character.propTypes = {
        name: PropTypes.string,
        height: PropTypes.string,
        mass: PropTypes.string,
        birth_year: PropTypes.string,
        gender: PropTypes.string
}

export default Character;