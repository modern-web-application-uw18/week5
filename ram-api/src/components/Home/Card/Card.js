import React from 'react';
import './Card.css';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import MdStar from 'react-icons/lib/md/star';

const card = (props)=> {


    


        return (
            <NavLink to={'/detail/' + props.character.id} exact={true}>
                <div className="Card" style={{ backgroundImage: `url(${props.character.image})` }}>
                    <span>
                        {JSON.parse(localStorage.getItem("favoriteIds")).indexOf(props.character.id) !== -1 ? <MdStar className="CardStar"/> : false}
                    </span>
                    <span className="Name">
                        {props.character.name}

                    </span>
                </div>
            </NavLink>


        )
    
}

card.propTypes = {
    character: PropTypes.object.isRequired
}

export default card;