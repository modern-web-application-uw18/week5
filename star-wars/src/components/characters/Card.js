import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Card.css';

class Card extends Component {

  render() {
    const url = `characters/${this.props.data.url.split('/')[5]}`;

    return (
      <div>
        <Link className='character-list' to={url}>{this.props.data.name}</Link>
      </div>
    );
  }
}

Card.propTypes = {
  data: PropTypes.shape({
      name: PropTypes.string,
      birth_year: PropTypes.string
  })
};

export default Card;
