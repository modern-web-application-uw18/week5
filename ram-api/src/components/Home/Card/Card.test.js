import React from 'react';
import { mount } from 'enzyme';
import Card from './Card'
import { MemoryRouter } from 'react-router-dom';
import character from '../../../__mocks__/character.json'

describe('<Card />', () => {
    it('Should render Card', () => {

        const wrapper = mount(
            <MemoryRouter>
                <Card character={character} />
            </MemoryRouter>
        );


        expect(wrapper).toMatchSnapshot();
    })
})




