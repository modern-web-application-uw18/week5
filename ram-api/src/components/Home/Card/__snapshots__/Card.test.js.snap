// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Should render Card 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "fugdr8",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "fugdr8",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <card
      character={
        Object {
          "created": "2017-11-04T19:09:56.428Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/6",
            "https://rickandmortyapi.com/api/episode/7",
            "https://rickandmortyapi.com/api/episode/8",
            "https://rickandmortyapi.com/api/episode/9",
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/11",
            "https://rickandmortyapi.com/api/episode/12",
            "https://rickandmortyapi.com/api/episode/14",
            "https://rickandmortyapi.com/api/episode/15",
            "https://rickandmortyapi.com/api/episode/16",
            "https://rickandmortyapi.com/api/episode/17",
            "https://rickandmortyapi.com/api/episode/18",
            "https://rickandmortyapi.com/api/episode/19",
            "https://rickandmortyapi.com/api/episode/20",
            "https://rickandmortyapi.com/api/episode/21",
            "https://rickandmortyapi.com/api/episode/22",
            "https://rickandmortyapi.com/api/episode/23",
            "https://rickandmortyapi.com/api/episode/24",
            "https://rickandmortyapi.com/api/episode/25",
            "https://rickandmortyapi.com/api/episode/26",
            "https://rickandmortyapi.com/api/episode/27",
            "https://rickandmortyapi.com/api/episode/29",
            "https://rickandmortyapi.com/api/episode/30",
            "https://rickandmortyapi.com/api/episode/31",
          ],
          "gender": "Female",
          "id": 3,
          "image": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Summer Smith",
          "origin": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "species": "Human",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/3",
        }
      }
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        exact={true}
        to="/detail/3"
      >
        <Route
          exact={true}
          path="/detail/3"
        >
          <Link
            aria-current={false}
            replace={false}
            to="/detail/3"
          >
            <a
              aria-current={false}
              href="/detail/3"
              onClick={[Function]}
            >
              <div
                className="Card"
                style={
                  Object {
                    "backgroundImage": "url(https://rickandmortyapi.com/api/character/avatar/3.jpeg)",
                  }
                }
              >
                <span />
                <span
                  className="Name"
                >
                  Summer Smith
                </span>
              </div>
            </a>
          </Link>
        </Route>
      </NavLink>
    </card>
  </Router>
</MemoryRouter>
`;
